### PR TITLE
Common tests

### PR DESCRIPTION
**Pull request by arcusfelis**
In this PR we:
- add Common Tests (Erlang testing framework) for Helm install/upgrade operations.
- We setup PgSQL.

Why Helm Charts for PG?
- do not wanna use both Docker and k8s...
- We wanna run tests locally, so not CircleCI way of starting PG.
- I wanted to try PG charts, so I can learn what is missing in MongooseIM charts.

Feedback for PG charts:
- I've tried it, and they are a bit weird. For example, there is no sane way to insert schema. I am using psql + `kubectl exec` till I figure out how to provide them as options.

Later in separate PRs we could:
- Test with MySQL.
- Test netsplits or node goes down case.
- Test that data is replicated and not lost after upgrade?
- Add escalus tests? Or AMOC if it makes sense. But client interruction smoke tests.

You can run locally using:
```
rebar3 ct
```

You need kind though.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -45,6 +49,55 @@ jobs:
...
             git push origin packages --force
+  k8s-with-kind:
+    executor: otp_26
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Install KinD
+          command: |
+            curl -Lo kind https://github.com/kubernetes-sigs/kind/releases/download/v0.20.0/kind-linux-amd64
```
Comment (NelsonVides) [inline .circleci/config.yml:60]: Would be good to have those version numbers as either yaml references as above, or environment variables to this job. Makes it more obvious how to update later.

File: .circleci/config.yml [hunk 2]
```diff
@@ -45,6 +49,55 @@ jobs:
...
+            chmod +x rebar3
+            sudo mv rebar3 /usr/local/bin/
+      - run:
+          name: Create Kubernetes Cluster
+          command: |
+            kind create cluster --loglevel debug
+      - run:
+          name: Install kubectl
+          command: |
+            curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.27.3/bin/linux/amd64/kubectl
```
Comment (NelsonVides) [inline .circleci/config.yml:76]: Version number as mentioned before

File: test/test_wait.erl [hunk 3]
```diff
@@ -0,0 +1,74 @@
+-module(test_wait).
```
Comment (NelsonVides) [inline test/test_wait.erl:1]: Why don't we make this a package? We have copy-pasted the same code in a thousand places already xD

File: test/mim_kind_SUITE.erl [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+run_psql_query(Query) ->
+    run(psql(" -U postgres -d mongooseim -c \"" ++ Query ++ "\"")).
+
+cmd(Cmd) ->
+   cmd(Cmd, #{}).
+
+cmd(Cmd, Opts) ->
+    {ok, CWD} = file:get_cwd(),
+    PortOpts =
+        [{cd, filename:join(CWD, "../../../../")}, exit_status, use_stdio, binary]
```
Comment (NelsonVides) [inline test/mim_kind_SUITE.erl:95]: Magic path 🔥

File: test/mim_kind_SUITE.erl [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        ++ [stderr_to_stdout || maps:get(stderr_to_stdout, Opts, true)],
+    Port = erlang:open_port({spawn, Cmd}, PortOpts),
+    receive_loop(Cmd, Port, <<>>).
+
+run_json(Cmd) ->
+    {0, Text} = run(Cmd, #{stderr_to_stdout => false}),
+    JSON = jiffy:decode(Text, [return_maps]),
+    ct:pal("~p", [JSON]),
```
Comment (NelsonVides) [inline test/mim_kind_SUITE.erl:103]: Do we need this or is it a debug artifact? Doesn't bother much, but just asking.

**Review by NelsonVides — COMMENTED** [review]
Very good idea adding tests for this!

File: test/mim_kind_SUITE.erl (hunk 4 above)
Comment (arcusfelis) [reply test/mim_kind_SUITE.erl:95]: Context: CT changes current directory into the log_dir.

yep, sadly CT does not provide repo dir.
we can provide it using env vars, like in big tests, but we would have to wrap.

OR we can make some rebar plugin :D
But repo dir is always 4 layers down, so, no magic, just boring stuff.

Comment (NelsonVides) [reply test/mim_kind_SUITE.erl:95]: Doesn't matter much, just a comment, but thanks for the explanation 😄

**Review by NelsonVides — APPROVED** [review]

**Pull request merged**